data/snippet/hot_link_protection.toml

```diff
@@ -0,0 +1,33 @@
+name = "Hot-link Protection"
+description = "Block other websites from linking to your content"
+code = """
+async function handleRequest(request) {
+  // Fetch the request
+  let response = await fetch(request)
+  // If it's an image, engage hotlink protection based on the
+  // Referer header.
+  let referer = request.headers.get('Referer')
+  let contentType = response.headers.get('Content-Type') || ''
+  if (referer && contentType.startsWith('image/')) {
+    // It's an image and there's a Referer. Verify that the
+    // hostnames match.
+    if (new URL(referer).hostname !==
+        new URL(request.url).hostname) {
+      // Hosts don't match, so this is a hotlink.
+      // Redirect the user to your website
+      return Response.redirect(url, 302)
+    }
+  }
+  // Everything is fine, return the response normally.
+  return response
+}
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+/**
+ * The url to redirect to
+ * @param {String} url
+*/
+const url = 'https://tutorial.cloudflareworkers.com/'
```

Comment:
```suggestion
const HOMEPAGE_URL = 'https://tutorial.cloudflareworkers.com/'
```

Reply:
Was unclear at first that this was just for `/images` maybe have this be a variable too? Or could add to the description
```
const PROTECTED_FOLDER = "/images"
```